quantum/debounce/sym_defer_pk.c

```diff
@@ -56,13 +56,9 @@ static void start_debounce_counters(matrix_row_t raw[], matrix_row_t cooked[], u
 
 // we use num_rows rather than MATRIX_ROWS to support split keyboards
 void debounce_init(uint8_t num_rows) {
-    debounce_counters = (debounce_counter_t *)malloc(num_rows * MATRIX_COLS * sizeof(debounce_counter_t));
-    int i             = 0;
-    for (uint8_t r = 0; r < num_rows; r++) {
-        for (uint8_t c = 0; c < MATRIX_COLS; c++) {
-            debounce_counters[i++] = DEBOUNCE_ELAPSED;
-        }
-    }
+    const size_t debounce_counters_size = num_rows * MATRIX_COLS * sizeof(debounce_counter_t);
+    debounce_counters                   = (debounce_counter_t *)malloc(debounce_counters_size);
+    memset(debounce_counters, DEBOUNCE_ELAPSED, debounce_counters_size);
```

Comment:
This `memset` can easily break if this code gets revisited later on. If the type of `debounce_counter_t` gets upgraded at the same time as the value of `DEBOUNCE_ELAPSED` becomes non-zero, both of which are abstracted and which a developer will reasonably assume can be changed without much fuss, the counters will then get initialized to the wrong value.
```suggestion
    debounce_counters = (debounce_counter_t *)malloc(num_rows * MATRIX_COLS * sizeof(debounce_counter_t));
    for (size_t i = 0; i <= num_rows * MATRIX_COLS; i++) {
        debounce_counters[i] = DEBOUNCE_ELAPSED;
    }
```

Reply:
I'm not sure I understand what you mean. Unless debounce_counters changes to being a pointer of pointers?

Reply:
I meant more like changing it to `uint16_t`. Since `memset` works on `char`, if the type `debounce_counter_t` gets changed to `uint16_t`, with a `DEBOUNCE_ELAPSED` value of 1, `memset` will make the elements 257 (0x0101) rather than 1.

Reply:
Got it. I will change it to a loop then.